sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -269,7 +269,8 @@ case class FileSourceScanExec(
   }
 
   @transient
-  private val pushedDownFilters = dataFilters.flatMap(DataSourceStrategy.translateFilter)
+  private val pushedDownFilters = dataFilters.flatMap(DataSourceStrategy.
+    translateFilter(_, !relation.fileFormat.isInstanceOf[ParquetSource]))
```

Comment:
I don't think we accept changes like this. If this is specific to parquet, do it in `ParquetFilters`.

And I still prefer to normalize the filters and remove unnecessary cast, before pushing filters down to data sources.

Reply:
Thanks
 Yeah, this is not a good solution, I can't solve this problem better now.